lower.ts

```diff
@@ -73,7 +73,7 @@ function lowerClass(cls: AST.Class<Type>, env : GlobalEnv) : IR.Class<Type> {
 function literalToVal(lit: AST.Literal) : IR.Value<Type> {
     switch(lit.tag) {
         case "num":
-            return { ...lit, value: BigInt(lit.value) }
+            return { ...lit, value: BigInt((lit.value).toLocaleString('fullwide', {useGrouping:false})) }
```

Comment:
I don't understand this line – could you add a comment or explain in more detail why it's needed? What's a test that would fail without this line and why?

Reply:
There is an unexpected error when passing the number directly to BigInt.
```
> BigInt(1e23)
99999999999999991611392n
> BigInt(100000000000000000000000)
99999999999999991611392n
> BigInt("100000000000000000000000")
100000000000000000000000n
```